src/runtime.rs

```diff
@@ -215,13 +216,13 @@ fn parse_build_args_or_exit() -> BuildArgs {
 
 fn read_buildpack_dir<E: Display + Debug>() -> Result<PathBuf, E> {
     env::var("CNB_BUILDPACK_DIR")
-        .map_err(Error::CannotDetermineBuildpackDirectory)
+        .map_err(|e| LibError::CannotDetermineBuildpackDirectory(e).into())
         .map(PathBuf::from)
 }
 
 fn read_buildpack_toml<BM: DeserializeOwned, E: Display + Debug>() -> Result<BuildpackToml<BM>, E> {
     read_buildpack_dir().and_then(|buildpack_dir| {
         read_toml_file(buildpack_dir.join("buildpack.toml"))
-            .map_err(Error::CannotReadBuildpackDescriptor)
+            .map_err(|e| LibError::CannotReadBuildpackDescriptor(e).into())
```

Comment:
I want to try to explain this back. 


Before we were calling

```
.map_err(Error::CannotReadBuildpackDescriptor)
```

Before this was defined here:

```rs
#[derive(thiserror::Error, Debug)]
pub enum Error<E: Debug + Display> {
	// ...
    #[error("Cannot read buildpack descriptor (buildpack.toml): {0}")]
    CannotReadBuildpackDescriptor(TomlFileError),
	// ...
}
```

Now it is defined as:

```rs
#[derive(thiserror::Error, Debug)]
pub enum Error<E: Debug + Display> {
    #[error("libcnb error: {0}")]
    LibError(LibError),

    // ...
} 


/// An error that occurred from libcnb
#[derive(thiserror::Error, Debug)]
pub enum LibError {
	//...
    #[error("Cannot read buildpack descriptor (buildpack.toml): {0}")]
    CannotReadBuildpackDescriptor(TomlFileError),
    // ...
}
```

And we are calling:

```rs
             .map_err(|e| LibError::CannotReadBuildpackDescriptor(e).into())
```

This is needed because before the error was passed to `Error::CannotReadBuildpackDescriptor` and it was formatted with:

```rs
    #[error("Cannot read buildpack descriptor (buildpack.toml): {0}")]
```

And it's bound by generic Display + Debug and also by `libcnb::Result`

```rs
pub type Result<T, E> = std::result::Result<T, Error<E>>;
```

Because we're using a concrete error enum, we cannot just pass the error to the enum. We've also got to tell rust how to convert it to generic `Error<E>`. And the way to do that is calling `into()`. Adding this syntax means we cannot implicitly pass the error and have to switch to using `|e|` syntax as well.

I'm not sure what code allows the LibError to be coerced into a:

```rs
#[derive(thiserror::Error, Debug)]
pub enum Error<E: Debug + Display> {
	//...
}
```

On second glance it looks like it's here:

```rs
impl<E: Debug + Display> From<LibError> for Error<E> {
    fn from(error: LibError) -> Self {
        Self::LibError(error)
    }
}
```

Which tells Error<E> how to convert LibError into Error<E> and this is what gets invoked when `into()` is called. https://doc.rust-lang.org/std/convert/trait.Into.html

Let me know if that's correct or if anything in my understanding needs to be adjusted.